app/models/piece.rb

```diff
@@ -127,10 +120,15 @@ def own_piece?(piece)
     piece.color == color
   end
 
-  def position_exist?(x, y)
-    x >= 0 && x <= 7 && 
-      y >= 0 && y <= 7
+  def final_position_exist?(x, y)
+    x != nil && y != nil && x >= 0 && x <= 7 && y >= 0 && y <= 7
```

Comment:
ahh i see, you want to check if the piece is on the board not just if the position is on the board. 

check out my valid move refactor PR from 2 days ago.  I will push a commit in 10 mintues with the change to check if the piece is on the board as you did here.